### ISSUE TITLE
Macro generated by other macro doesn't expand fully

### ISSUE THREAD

**Issue by jmgawecki** [apple/swift-syntax#2629]
### Description

The `CoreDataManaged` (memberAttribute and a few others that aren't relevant here) macro
![Screenshot 2024-04-27 at 07 15 40](https://github.com/apple/swift-syntax/assets/71930261/2e235797-3c2f-40d3-ace2-6d74a6b08540)
Expands to following, where `CoreDataArray` is now attached to the property `entities`
![Screenshot 2024-04-27 at 07 17 32](https://github.com/apple/swift-syntax/assets/71930261/a6c35105-5dff-4ddb-b174-60062c0a8d64)
Then, `CoreDataArray` (peer, accessor) macro expands but not fully, the peer expansion is missing from the viewer but I can confirm that the property has been in fact generated and and can be accessed.
![Screenshot 2024-04-27 at 07 18 11](https://github.com/apple/swift-syntax/assets/71930261/58633c4d-ae99-4952-881d-b22476b360ac)

This is what happens when I attach the `CoreDataArray` myself: 
![Screenshot 2024-04-27 at 07 20 26](https://github.com/apple/swift-syntax/assets/71930261/6de8b45e-f3fe-4986-a2ff-1a5321935dd3)

This looks like a bug to me when the macro expansions are interfering with each other somehow.
This was noted on both Swift 6.0 snapshot and Swift 5.10 release version running Xcode 15.3

**Comment by ahoppen** [comment]
Synced to Apple’s issue tracker as rdar://127156691

**Comment by ahoppen** [comment]
This is an issue in Xcode, not with swift-syntax or the compiler. I just verified that sourcekitd (which provides semantic that raw expansion information to Xcode and lives in the compiler repo) does return the nested expansions, so I’m closing the issue.

I did pass the issue on to the Xcode team. If you are interested in seeing the status of that issue, I suggest that you file a feedback at https://feedbackassistant.apple.com.
